content/usgs-packages/dataRetrieval_Discovery.Rmd

```diff
@@ -0,0 +1,296 @@
+
+`whatNWISdata` will return a data.frame specifying the types of data available for a specified site(s) that fits your querying criteria. You can add queries by the data service, USGS parameter code, or statistics code. Continuing with the South Carolina temperature data example from the `whatNWISsites` section,  let's look for the mean daily stream temperature. Since you cannot query `whatNWISdata` using `siteType` or `stateCd`, let's use the site numbers from the previous lesson to specify South Carolina stream sites. You will need to specify the parameter again since the sites likely collect more than one parameter.
+
+```{r data_sc_stream_temp_avg, message=FALSE, warning=FALSE}
+# Average daily SC stream temperature data
+data_sc_stream_temp_avg <- whatNWISdata(
+  siteNumbers = unique(sites_sc_stream_temp[['site_no']]),
+  parameterCd="00010",
+  service="dv",
+  statCd="00003")
+nrow(data_sc_stream_temp_avg)
+```
+
+In some queries, you might notice an error that says "Too Long". This means that you are querying for more data than `whatNWISdata` can handle. If this happens, try shortening the number of sites you are querying at a time, and append all results at the end.
+
+The data returned from this query can give you information about the data available for each site including, date of first and last record (`begin_date`, `end_date`), number of records (`count_nu`), site altitude (`alt_va`), corresponding hydrologic unit code (`huc_cd`), and parameter units (`parameter_units`). These columns allow even more specification of data requirements before actually downloading the data. 
+
+To illustrate this, let's apply an additional filter to these data using the `filter` function from `dplyr`. Imagine that the trend analysis you are conducting requires a minimum of 300 records and the most recent data needs to be no earlier than 1975.
+
+```{r data_sc_stream_temp_avg_applicable, message=FALSE, warning=FALSE}
+# Useable average daily SC stream temperature data
+library(dplyr)
+data_sc_stream_temp_avg_applicable <- data_sc_stream_temp_avg %>% 
+  filter(count_nu >= 300, end_date >= as.Date("1975-01-01"))
+nrow(data_sc_stream_temp_avg_applicable)
+```
+
+This means you would have `r prettyNum(nrow(data_sc_stream_temp_avg_applicable), big.mark=",")` of sites to work with for your study.
+
+## Common WQP function arguments
+
+*_`countrycode`_*, *_`statecode`_*, and *_`countycode`_* These geopolitical filters must be specified by Federal Information Processing Standard (FIPS) codes, not names. Each FIPS code is a two letter indicator and must be preceded by the FIPS code of the larger geopolitical filter. For example, the FIPS code for the United States is `US`, and the FIPS code for South Carolina is `45`. When querying with the statecode, I would enter `statecode="US:45"`. The same rule extends to county FIPS. There is a built-in R data frame called state.fips that can help you determine the correct state code, otherwise, you can search online.
```

Comment:
Oh, country, huh, didn't realize that was an option. "US:" is tacked onto the state and county arguments in the lookup functions. If it's outside the US, you'll need to do the proper "MX:foo:bar" or whatever.